agent/agent-plugins/redis-jedis-2.x/src/main/java/org/bithon/agent/plugin/redis/jedis2/interceptor/OnCommand.java

```diff
@@ -50,7 +49,7 @@ public class OnCommand extends AroundInterceptor {
     public InterceptionDecision before(AopContext aopContext) throws Exception {
         Jedis jedis = aopContext.getTargetAs();
         String hostAndPort = jedis.getClient().getHost() + ":" + jedis.getClient().getPort();
-        long db = jedis.getDB();
+        long db = jedis.getClient().getDB();
```

Comment:
how did the original code cause the NoClassDefFoundError?

Reply:
The change looks good to me because Line 51 also uses the `getClient` method to access some information.

what I don't understand is that inside the `jedis.getDB()`, it calls `getClient().getDB()`, which looks the same as `jedis.getClient().getDB()`.

Could you upload your test project to GitHub so that I can investigate more if there's sth we can improve inside the class loader?